crates/solver/src/settlement_submission/flashbots_settlement.rs

```diff
@@ -97,10 +97,19 @@ impl<'a> FlashbotsSolutionSubmitter<'a> {
                 .unwrap_or_else(|_| Duration::from_secs(0)),
         );
 
-        futures::select! {
-            method_error = submit_future.fuse() => tracing::info!("stopping submission because simulation failed: {:?}", method_error),
-            new_nonce = nonce_future.fuse() => tracing::info!("stopping submission because account nonce changed to {}", new_nonce),
-            _ = deadline_future.fuse() => tracing::info!("stopping submission because deadline has been reached"),
+        let fallback_result = tokio::select! {
+            method_error = submit_future.fuse() => {
+                tracing::info!("stopping submission because simulation failed: {:?}", method_error);
+                Err(method_error)
+            },
+            new_nonce = nonce_future.fuse() => {
+                tracing::info!("stopping submission because account nonce changed to {}", new_nonce);
+                Ok(None)
+            },
+            _ = deadline_future.fuse() => {
+                tracing::info!("stopping submission because deadline has been reached");
+                Ok(None)
```

Comment:
Aren't deadlines errors in regular transaction submission?

Reply:
They are. Ok(None) will cause an error message. The only case where we don't error is if we return Err(e) and e can be downcasted to a method error.